### PR TITLE
Implement CodeAct pattern with Gemini native code execution

### PR DESCRIPTION
**Pull request by srmsoumya**
Invalidates #480 Addresses #487

Replaces prompt-based LLM insight generation with CodeAct using Gemini's native code executor.  

### Changes
- New `GeminiCodeExecutor` for stateless code execution with inline CSV data
- Agent explores datasets, writes Python code to analyze data, and generates chart outputs
- Updated graph state to track code execution details
- Added frontend UI support in streamlit to display code blocks and execution outputs 

All tests for generate-insights are passing & I have tested a few queries on streamlit. This will need a bit of prompt engineering to handle all edge-cases before we can deploy to production.

cc @sunu @batpad @geohacker

### CODE REVIEW

File: src/tools/generate_insights.py [hunk 1]
```diff
@@ -228,143 +179,158 @@ async def generate_insights(
...
+Generate ALL content (insights, titles, follow-ups) in the SAME LANGUAGE as the user query.
+Dataset specific guidelines: {dataset_guidelines}
+Dataset specific cautions: {dataset_cautions}
 
-        # Add follow-up suggestions
-        message_parts.append("Follow-up suggestions:")
-        for i, suggestion in enumerate(follow_ups, 1):
-            message_parts.append(f"{i}. {suggestion}")
+#### Capability Context
+You can analyze data for any area of interest, pull data from datasets like {available_datasets} for different time periods, and create various charts/insights. Base follow-ups on what's actually possible with available data and tools.
```
Comment (yellowcap) [inline src/tools/generate_insights.py]: this does not make sense to me at this point. We are not pulling data here at all.

**Review by yellowcap — COMMENTED** [review]
A few comments from initial read, I'll test this locally now.
